antarest/study/storage/patch_service.py

```diff
@@ -1,28 +1,48 @@
 from pathlib import Path
-from typing import Union
+from typing import Union, Optional
+
+from pydantic import ValidationError
 
-from antarest.core.model import JSON
 from antarest.study.model import Patch, PatchOutputs, RawStudy
+from antarest.study.repository import StudyMetadataRepository
 from antarest.study.storage.rawstudy.model.filesystem.factory import FileStudy
 from antarest.study.storage.variantstudy.model.dbmodel import (
     VariantStudy,
 )
 
 
 class PatchService:
-    def get(self, study: Union[RawStudy, VariantStudy, FileStudy]) -> Patch:
-        patch_path = (
-            Path(study.path)
-            if not isinstance(study, FileStudy)
-            else study.config.study_path
-        ) / "patch.json"
+    def __init__(self, repository: Optional[StudyMetadataRepository] = None):
+        self.repository = repository
+
+    def get(
+        self, study: Union[RawStudy, VariantStudy], get_from_file: bool = False
+    ) -> Patch:
+        patch = Patch()
+
+        if not get_from_file:
+            try:
+                patch = Patch.parse_raw(study.additional_data.patch)
+            except (AttributeError, ValidationError):
+                get_from_file = True
+
+        if get_from_file:
+            patch_path = (Path(study.path)) / "patch.json"
+            if patch_path.exists():
+                patch = Patch.parse_file(patch_path)
+
+        return patch
+
+    def get_from_filestudy(self, file_study: FileStudy) -> Patch:
+        patch = Patch()
+        patch_path = (Path(file_study.config.study_path)) / "patch.json"
         if patch_path.exists():
-            return Patch.parse_file(patch_path)
-        return Patch()
+            patch = Patch.parse_file(patch_path)
+        return patch
```

Comment:
why this method ? the previous one already do that

Reply:
For typing purpose, the other one uses a Study, this one a FileStudy

Reply:
oh.. ok I missed the type.